src/Stage/Parse/AdvancedWithState.elm

```diff
@@ -0,0 +1,455 @@
+module Stage.Parse.AdvancedWithState exposing
+    ( DeadEnd
+    , Nestable(..)
+    , Parser
+    , Step(..)
+    , Token(..)
+    , Trailing(..)
+    , andThen
+    , backtrackable
+    , chompIf
+    , chompUntil
+    , chompUntilEndOr
+    , chompWhile
+    , commit
+    , end
+    , float
+    , getChompedString
+    , getCol
+    , getIndent
+    , getOffset
+    , getPosition
+    , getRow
+    , getSource
+    , ignore
+    , inContext
+    , int
+    , keep
+    , keyword
+    , lazy
+    , lineComment
+    , loop
+    , map
+    , mapChompedString
+    , multiComment
+    , number
+    , oneOf
+    , problem
+    , run
+    , sequence
+    , spaces
+    , succeed
+    , symbol
+    , token
+    , updateState
+    , variable
+    , withIndent
+    , withState
+    )
+
+{-| Everything here works just like in the
+[`Parser.Advanced`](/packages/elm/parser/latest/Parser-Advanced)
+module, except that there are no infixes `(|=)` and `(|.)`.
+
+You can use the pipe operator and the `keep` and `ignore` functions,
+respectively, to replace them. Eg.:
+
+    succeed identity
+        |> keep myParser
+        |> ignore spaces
```

Comment:
I believe I've seen `keep / skip` naming somewhere, which mighit be nicer because the two functions have names of the same length? WDYT?

Reply:
I agree it is nicer, but I think it is harder to distinguish each one, eg.:

    |> P.keep moduleType
    |> P.skip spacesCommentsAndGreaterIndent
    |> P.keep moduleName
    |> P.skip spacesCommentsAndGreaterIndent
    |> P.skip (P.keyword (P.Token "exposing" ExpectingExposingKeyword))
    |> P.skip spacesCommentsAndGreaterIndent
    |> P.keep exposingList

With the `ignore`:

    |> P.keep moduleType
    |> P.ignore spacesCommentsAndGreaterIndent
    |> P.keep moduleName
    |> P.ignore spacesCommentsAndGreaterIndent
    |> P.ignore (P.keyword (P.Token "exposing" ExpectingExposingKeyword))
    |> P.ignore spacesCommentsAndGreaterIndent
    |> P.keep exposingList

Reply:
Ah, got it. The same length might actually be a disadvantage, not an advantage. Cool to see it on specific code snippets.